### PR TITLE
Remove stdlib stuff from the Reference

### LINKED ISSUE #11794: remove standard library information from the Rust manual
I think this should be a language specification, and should avoid describing the library. The language semantics and grammar are a much saner scope to cover than the entire standard library. Implementation details like the standard library's choice of allocator are even more out of place.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #11794

I mostly removed superflous examples which use the standard library.

I have one more quesiton here though: threads. They're mostly a library thing, at this point, right?

### CODE REVIEW

**Comment by steveklabnik** [comment]
@nikomatsakis I re-added in `Copy`, tweaked that wording, and removed references to threads.

File: src/doc/reference.md [hunk 1]
```diff
@@ -3882,75 +3811,17 @@ impl Printable for String {
...
-
-## Memory model
-
-A Rust program's memory consists of a static set of *items*, a set of
-[threads](#threads) each with its own *stack*, and a *heap*. Immutable portions of
-the heap may be shared between threads, mutable portions may not.
+# The `Copy` trait
+
+Rust has a special trait, `Copy`, which when implemented changes the semantics
+of a value. Values whos type implements `Copy` are copied rather than moved
```
Comment (nikomatsakis) [inline src/doc/reference.md]: Typo: "whose"

File: src/doc/reference.md [hunk 2]
```diff
@@ -3882,75 +3811,17 @@ impl Printable for String {
...
-    All values of this kind can be implicitly copied.
-    This kind includes scalars and immutable references,
-    as well as structural types containing other `Copy` types.
-* `'static`
-  : Types of this kind do not contain any references (except for
-    references with the `static` lifetime, which are allowed).
-    This can be a useful guarantee for code
-    that breaks borrowing assumptions
-    using [`unsafe` operations](#unsafe-functions).
-* `Drop`
```
Comment (nikomatsakis) [inline src/doc/reference.md]: The `Drop` and `Sized` traits are also very "well-known" to the compiler -- probably worth keeping a mention of those.

File: src/doc/reference.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/reference.md]: fixed

File: src/doc/reference.md [hunk 3]
```diff
@@ -3972,25 +3839,11 @@ in the heap, heap allocations may outlive the frame they are allocated within.
...
-as boxes and references.
-
-When a thread sends a value that has the `Send` trait to another thread, it loses
-ownership of the value sent and can no longer refer to it. This is statically
-guaranteed by the combined use of "move semantics", and the compiler-checked
-_meaning_ of the `Send` trait: it is only instantiated for (transitively)
-sendable kinds of data constructor and pointers, never including references.
-
 When a stack frame is exited, its local allocations are all released, and its
 references to boxes are dropped.
```
Comment (nikomatsakis) [inline src/doc/reference.md]: this should probably say "destructors are executed" but anyway

File: src/doc/reference.md (hunk 2 above)
Comment (steveklabnik) [inline src/doc/reference.md]: seems good, small notes added

File: src/doc/reference.md (hunk 3 above)
Comment (steveklabnik) [inline src/doc/reference.md]: yeah we can go over this with the general reference cleanup post-beta

**Comment by steveklabnik** [comment]
@nikomatsakis ping!

**Pull request merged**